File: python/7 kyu/306.py
# https://www.codewars.com/kata/69e8cd7f48dd4ffb1d25ef2f/train/python

# Passed

def rumor_starter(record):
    all_people_who_were_told_the_rumor_set = set()
    for people_who_were_told_the_rumor_arr in record.values():
        all_people_who_were_told_the_rumor_set.update(people_who_were_told_the_rumor_arr)
    
    people_who_started_the_rumor = list()
    for person in record:
        if person not in all_people_who_were_told_the_rumor_set:
            people_who_started_the_rumor.append(person)
            
    people_who_started_the_rumor.sort()
    return people_who_started_the_rumor

output = rumor_starter({"Haley":["Steph"], "Josh":["Ava"], "Kate":["Ava","Josh","Mia"], "Kim":["Ava","Josh","Mia"], "Ava":[], "Elly":["Kim"], "Mia":[]})
print(output)

File: nodejs/7 kyu/306.js
// https://www.codewars.com/kata/58941fec8afa3618c9000184/train/javascript

// Passed

function growingPlant(upSpeed, downSpeed, desiredHeight) {
  let plantHeight = 0;
  let days = 0;
  while (plantHeight < desiredHeight) {
    days += 0.5;
    if ((days / 0.5) % 2 === 1) {
      plantHeight += upSpeed;
    } else {
      plantHeight -= downSpeed;
    }
  }

  days = Math.ceil(days);
  return days;
}

const output = growingPlant(100, 10, 910);
console.log(output);